src/main/java/by/bookify/image/service/GridFsMongoImageService.java

```diff
@@ -0,0 +1,66 @@
+package by.bookify.image.service;
+
+import by.bookify.image.exception.ImageNotFoundException;
+import by.bookify.image.exception.ImageUploadException;
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBObject;
+import com.mongodb.client.gridfs.model.GridFSFile;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.gridfs.GridFsTemplate;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class GridFsMongoImageService implements ImageService {
+
+    private GridFsTemplate gridFsTemplate;
+
+    @Override
+    public void uploadImage(Long bookId, MultipartFile file) {
+        DBObject metaData = createBasicDBObject(bookId);
+        gridFsTemplate.delete(new Query(Criteria.where("metadata.id").is(bookId)));
```

Comment:
You have used metadata.id twice. Is it a constant value?

Reply:
okey man